### PR TITLE
NeuronType functions handle array_like manually

### PR DESCRIPTION
**Pull request by tbekolay**
**Motivation and context:**
This is an alternative to #633, which achieved the same ends with manual `np.array` calls instead of a decorator. The consensus in dev meetings has been to go with the manual approach, so I wanted to make this to ensure that it still seems better to most people before closing #633.

**Interactions with other PRs:**
Same functionality as #633, let's not merge both!

**How has this been tested?**
Tried this in a terminal:

```
>>> nengo.LIF().gain_bias([100, 120], [-0.5, 0.5])
(array([ 1.35549652,  5.36847925]), array([ 1.67774826, -1.68423963]))
```

Which would have failed previously. Didn't add any more tests, but I could if people feel strongly that they should be added.

**How long should this take to review?**

- Quick (less than 40 lines changed or changes are straightforward)

**Types of changes:**

- Bug fix (non-breaking change which fixes an issue)

**Checklist:**

- [x] I have read the **CONTRIBUTING.rst** document.
- [x] I have updated the documentation accordingly.
- [NA] I have included a changelog entry.
- [NA] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: nengo/neurons.py [hunk 1]
```diff
@@ -42,18 +59,21 @@ def gain_bias(self, max_rates, intercepts):
...
         Returns
         -------
-        gain : ndarray(dtype=float64)
+        gain : (n_neurons,) array_like
             Gain associated with each neuron. Sometimes denoted alpha.
-        bias : ndarray(dtype=float64)
+        bias : (n_neurons,) array_like
             Bias current associated with each neuron.
         """
+        max_rates = np.array(max_rates, dtype=float, copy=False, ndmin=1)
```
Comment (tbekolay) [inline nengo/neurons.py]: Yep, that's right. I don't know if the `ndmin` is super necessary, but @hunse included it in his PR so I trust he did it for a reason!

**Comment by hunse** [comment]
LGTM. I'm fine with this instead of #633

**Comment by Seanny123** [comment]
This LGTM too. Should a test be made to check that `array_like` is handled properly or would that be silly to test?

**Comment by tbekolay** [comment]
> Should a test be made to check that array_like is handled properly or would that be silly to test?

Not silly, but also not necessary. If you want to write it up then by all means! But I would say that if someone wants to merge this they shouldn't wait for the example.

**Comment by jgosmann** [comment]
Two people said it looks good to them. Merge before the release, @tbekolay?

**Comment by tbekolay** [comment]
OK, will do.

**Pull request merged**